Inferences/BackwardDemodulation.cpp

```diff
@@ -76,13 +76,15 @@ struct BackwardDemodulation::RemovedIsNonzeroFn
 
 struct BackwardDemodulation::RewritableClausesFn
 {
-  RewritableClausesFn(DemodulationSubtermIndex* index) : _index(index) {}
+  RewritableClausesFn(DemodulationSubtermIndex* index, Literal* lit) : _index(index), _lit(lit) {}
   VirtualIterator<pair<TermList,TermQueryResult> > operator() (TermList lhs)
   {
-    return pvi( pushPairIntoRightIterator(lhs, _index->getInstances(lhs, true)) );
+    TermList sort = SortHelper::getTermSort(lhs, _lit);
```

Comment:
@joe-hauns , the above displays the same inefficiency you spotted in `TermIndex`. Whatever solution you apply there, should be applied here as well.

Reply:
Sorted them all out.